src/pages/workspace/WorkspaceInviteMessagePage.js

```diff
@@ -71,180 +71,151 @@ const defaultProps = {
 };
 
-class WorkspaceInviteMessagePage extends React.Component {
-    constructor(props) {
-        super(props);
+function WorkspaceInviteMessagePage(props) {
+    const focusTimeoutRef = useRef();
+    const welcomeMessageInputRef = useRef(null);
 
-        this.sendInvitation = this.sendInvitation.bind(this);
-        this.validate = this.validate.bind(this);
-        this.openPrivacyURL = this.openPrivacyURL.bind(this);
-        this.debouncedSaveDraf = _.debounce((newDraft) => {
-            Policy.setWorkspaceInviteMessageDraft(this.props.route.params.policyID, newDraft);
-        }, 2000);
-        this.state = {
-            welcomeNote: this.props.workspaceInviteMessageDraft || this.getDefaultWelcomeNote(),
-        };
-    }
-
-    componentDidMount() {
-        if (_.isEmpty(this.props.invitedEmailsToAccountIDsDraft)) {
-            Navigation.goBack(ROUTES.WORKSPACE_INVITE.getRoute(this.props.route.params.policyID), true);
-            return;
-        }
-        this.focusWelcomeMessageInput();
-    }
+    useEffect(
+        () => () => {
+            clearTimeout(focusTimeoutRef.current);
+        },
+        [],
+    );
 
-    componentDidUpdate(prevProps) {
-        if (!prevProps.isFocused && this.props.isFocused) {
-            this.focusWelcomeMessageInput();
-        }
+    const focusWelcomeMessageInput = () => {
+        focusTimeoutRef.current = setTimeout(() => {
+            welcomeMessageInputRef.current.focus();
+            // Below condition is needed for web, desktop and mweb only, for native cursor is set at end by default.
+            if (welcomeMessageInputRef.current.value && welcomeMessageInputRef.current.setSelectionRange) {
+                const length = welcomeMessageInputRef.current.value.length;
+                welcomeMessageInputRef.current.setSelectionRange(length, length);
+            }
+        }, CONST.ANIMATED_TRANSITION);
+    };
 
-        if (
-            !(
-                (prevProps.preferredLocale !== this.props.preferredLocale || prevProps.policy.name !== this.props.policy.name) &&
-                this.state.welcomeNote === Localize.translate(prevProps.preferredLocale, 'workspace.inviteMessage.welcomeNote', {workspaceName: prevProps.policy.name})
-            )
-        ) {
-            return;
+    useEffect(() => {
+        if (_.isEmpty(props.invitedEmailsToAccountIDsDraft)) {
+            Navigation.goBack(ROUTES.WORKSPACE_INVITE.getRoute(props.route.params.policyID), true);
         }
-        this.setState({welcomeNote: this.getDefaultWelcomeNote()});
-    }
+        focusWelcomeMessageInput();
+    }, [props.invitedEmailsToAccountIDsDraft, props.route.params.policyID]);
 
-    componentWillUnmount() {
-        if (!this.focusTimeout) {
-            return;
-        }
-        clearTimeout(this.focusTimeout);
-    }
+    const saveDraft = (newDraft) => {
+        Policy.setWorkspaceInviteMessageDraft(props.route.params.policyID, newDraft);
+    };
```

Comment:
In old component it was debounced, so it didn't affect performance that much, and as I asked below, are you sure it's necessary to save draft here?

Reply:
In the current implementation I have to keep it this way and not use debounce to avoid lag when changing the input value. If we have to use debounce here I will try to rewrite the logic to support it, let me know.